internal/icinga2/client_api.go

```diff
@@ -149,6 +149,45 @@ func (client *Client) queryObjectsApiQuery(ctx context.Context, objType string,
 		})
 }
 
+// fetchIcingaAppStatus retrieves the global state of the IcingaApplication type via the /v1/status endpoint.
+func (client *Client) fetchIcingaAppStatus(ctx context.Context) (*IcingaApplication, error) {
+	response, err := client.queryObjectsApi(
+		ctx,
+		[]string{"/v1/status/IcingaApplication/"},
+		http.MethodGet,
+		nil,
+		map[string]string{"Accept": "application/json"})
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		_, _ = io.Copy(io.Discard, response)
+		_ = response.Close()
+	}()
+
+	type status struct {
+		Status struct {
+			IcingaApplication *IcingaApplication `json:"icingaapplication"`
+		} `json:"status"`
+	}
+
+	var results []status
+	err = json.NewDecoder(response).Decode(&struct {
+		Results *[]status `json:"results"`
+	}{&results})
+	if err != nil {
+		return nil, err
+	}
+
+	app := new(IcingaApplication)
+	if len(results) != 0 {
+		app = results[0].Status.IcingaApplication
+	}
+
+	return app, nil
```

Comment:
The function already has an `error` return value, so I suggest returning an error if that information is missing (which it never should) instead of returning a zero-initialized struct.